sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala

```diff
@@ -361,4 +352,87 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       assert(spark.sharedState.cacheManager.lookupCachedData(samePlan).isDefined)
     }
   }
+
+  test("SPARK-27248 refreshTable should recreate cache with same cache name and storage level") {
+
+    // This section tests when a table is cached with its qualified name but its is refreshed with
+    // its unqualified name.
+    withTempPath { path =>
+      withTempDatabase { db =>
+        withTable(s"$db.cachedTable") {
+          withCache(s"$db.cachedTable") {
+
+            // Create table 'cachedTable' in default db for testing purpose.
+            sql(s"CREATE TABLE $db.cachedTable(key STRING) USING PARQUET LOCATION '${path.toURI}'")
```

Comment:
Ur, @William1104 .
From the last commit, this become a wrong fix. If you use `USING` keyword, it becomes Spark file-based table instead of Hive table. Since this test suite is for Hive table testing, you should not use `USING`.

Reply:
Sorry about this. The SQL is update to:

- s"CREATE TABLE $db.cachedTable AS SELECT 1 AS key"
- "CREATE TABLE cachedTable AS SELECT 1 AS key"

I also modified the function `withCache` in SQLTestUtils to avoid exception masking. Before the modification, any exception throwing from finally block may mask the exception throwing from try block. I hope the created function `withFinallyBlock` does the right job.

Reply:
Hi @dongjoon-hyun,

I modified the test further with following assert statements to make sure the test is running against Hive tables. Be honest, I am not sure whether this assert statement is necessary or not. If it is trivial and is not a good idea, I will remove them. 

```
val isHive = spark.table(s"$db.cachedTable")
            .queryExecution
            .optimizedPlan
            .collectFirst { case _: HiveTableRelation => true }
            .getOrElse(false)
assert(isHive, s"$db.cachedTable must be a HiveTable")
```

In addition to that, table creation statements are also updated with keyword `USING HIVE` to 
```
sql(s"CREATE TABLE $db.cachedTable(id INT, val STRING) USING HIVE")
sql("CREATE TABLE cachedTable(id INT, val STRING) USING HIVE")
```
I hope it can make it clear the test is for HIVE tables.

Appreciate if you can help to review the PR again when you have time. Many thanks. 

Best regards,
William